src/test/kotlin/no/nav/syfo/identhendelse/IdenthendelseServiceSpek.kt

```diff
@@ -0,0 +1,145 @@
+                    val oldIdenter = kafkaIdenthendelseDTO.getInactivePersonidenter()
+
+                    populateDatabase(oldIdenter.first(), database)
+
+                    val oldIdentOccurrences = database.getIdentOccurenceCount(oldIdenter)
+                    oldIdentOccurrences shouldBeEqualTo 4
+
+                    runBlocking {
+                        identhendelseService.handleIdenthendelse(kafkaIdenthendelseDTO)
+                    }
+
+                    val newIdentOccurrences = database.getIdentOccurenceCount(listOf(newIdent))
+                    newIdentOccurrences shouldBeEqualTo 4
+                }
+
+                it("Skal oppdatere tabeller der gammel ident forekommer når person har fått ny ident") {
+                    val kafkaIdenthendelseDTO = generateKafkaIdenthendelseDTO(hasOldPersonident = true)
+                    val newIdent = kafkaIdenthendelseDTO.getActivePersonident()!!
+                    val oldIdenter = kafkaIdenthendelseDTO.getInactivePersonidenter()
+
+                    populateDatabase(
+                        oldIdent = oldIdenter.first(),
+                        database = database,
+                        updateAll = false,
+                    )
+
+                    val oldIdentOccurrences = database.getIdentOccurenceCount(oldIdenter)
+                    oldIdentOccurrences shouldBeEqualTo 2
+
+                    runBlocking {
+                        identhendelseService.handleIdenthendelse(kafkaIdenthendelseDTO)
+                    }
+
+                    val newIdentOccurrences = database.getIdentOccurenceCount(listOf(newIdent))
+                    newIdentOccurrences shouldBeEqualTo 2
+                }
+            }
+
+            describe("Unhappy path") {
+                it("Skal kaste feil hvis PDL ikke har oppdatert identen") {
+                    val kafkaIdenthendelseDTO = generateKafkaIdenthendelseDTO(
+                        personident = UserConstants.ARBEIDSTAKER_3_PERSONIDENTNUMBER,
+                        hasOldPersonident = true,
+                    )
+                    val oldIdent = kafkaIdenthendelseDTO.getInactivePersonidenter().first()
+
+                    populateDatabase(oldIdent, database)
+
+                    assertFailsWith(IllegalStateException::class) {
+                        identhendelseService.handleIdenthendelse(kafkaIdenthendelseDTO)
+                    }
+                }
+            }
+        }
+    }
+})
+
+fun populateDatabase(oldIdent: PersonIdentNumber, database: DatabaseInterface, updateAll: Boolean = true) {
+    val stoppunkt = generateDialogmotekandidatStoppunktPlanlagt(oldIdent, LocalDate.now())
+    val endring = generateDialogmotekandidatEndringStoppunkt(oldIdent)
+
+    database.connection.createDialogmotekandidatStoppunkt(
+        commit = true,
+        dialogmotekandidatStoppunkt = stoppunkt,
+    )
+    database.createDialogmotekandidatEndring(endring)
+
+    if (updateAll) {
```

Comment:
Hva betyr egentlig `updateAll`? 🤔

Reply:
`oppdaterAlle` :trollface: 🇳🇴 Neida!
Men ja, kan finne et bedre navn, er vel bare et flagg for å teste at man ikke har identen i alle tabellene, og at det likevel skal gå fint 👍🏼